influxql/iterator.go

```diff
@@ -259,9 +261,11 @@ func newAuxIteratorFields(seriesKeys SeriesList, opt IteratorOptions) auxIterato
 
 func (a auxIteratorFields) close() {
 	for _, f := range a {
+		f.mtx.Lock()
 		for _, itr := range f.itrs {
 			itr.Close()
 		}
+		f.mtx.Unlock()
```

Comment:
Can you move these lines inside `f.Close()` and then do a `Lock/`defer Unlock()` inside of there? The slice shouldn't be reaching into the iterator field's mutex.